packages/eslint-plugin/src/rules/member-ordering.ts

```diff
@@ -615,6 +615,39 @@ function getRank(
   return getRankOrder(memberGroups, orderConfig);
 }
 
+function changeToGroups(
+  memberSet: Member[],
+  memberType: MemberType[],
+  supportsModifiers: boolean,
+): Member[][] {
+  const groupedMembers: Member[][] = [];
+
+  const getPreviousRank = (): number | undefined => {
+    if (groupedMembers.at(-1)?.at(0) === undefined) {
+      return undefined;
+    }
+    return getRank(
+      groupedMembers[groupedMembers.length - 1][0],
+      memberType,
+      supportsModifiers,
+    );
+  };
+
+  memberSet.forEach((member, index) => {
+    if (getRank(member, memberType, supportsModifiers) === getPreviousRank()) {
+      groupedMembers.at(-1)?.push(member);
+    } else if (index === memberSet.length - 1) {
+      return;
+    } else if (
```

Comment:
[Refactor] Bailing out early is always nice to save time & make exit cases more clear. No unit tests fail when I try this:

```suggestion
    if (index === memberSet.length - 1) {
      return;
    }

    if (getRank(member, memberType, supportsModifiers) === getPreviousRank()) {
      groupedMembers.at(-1)?.push(member);
    } else if (
```